tests/regression/56-witness/46-top-bool-invariant.t

```diff
@@ -146,9 +146,53 @@ interval_set only:
         value: x <= (_Bool)1
         format: c_expression
 
+bitfield only:
+
+  $ goblint --enable witness.yaml.enabled --set witness.yaml.invariant-types '["location_invariant"]' --disable ana.int.def_exc --disable ana.int.interval --disable ana.int.enums --disable ana.int.congruence --disable ana.int.interval_set --enable ana.int.bitfield 46-top-bool-invariant.c --enable witness.invariant.inexact-type-bounds
+  [Info][Deadcode] Logical lines of code (LLoC) summary:
+    live: 2
+    dead: 0
+    total lines: 2
+  [Info][Witness] witness generation summary:
+    location invariants: 3
+    loop invariants: 0
+    flow-insensitive invariants: 0
+    total generation entries: 1
+
+  $ yamlWitnessStrip < witness.yml
+  - entry_type: invariant_set
+    content:
+    - invariant:
+        type: location_invariant
+        location:
+          file_name: 46-top-bool-invariant.c
+          line: 5
+          column: 3
+          function: main
+        value: (_Bool)0 <= x
+        format: c_expression
+    - invariant:
+        type: location_invariant
+        location:
+          file_name: 46-top-bool-invariant.c
+          line: 5
+          column: 3
+          function: main
+        value: (x & (_Bool)1) == (_Bool)0
```

Comment:
This is wrong for a top `_Bool`, but I'm having hard time understanding why.

The representation of top `int` is `0b?...?, (zs:-1, os:-1)`, but the representation of top `_Bool` is `0b0...0?, (zs:-1, os:1)`. These are not consistent: for `int` it has an infinite sequence of unknown bits (including higher bits that fit into `int`), but for `_Bool` it's one unknown bit with infinitely many definite zero bits in front.
I suspect this causes something to go wrong because the invariant generation is made to believe that there are all these definite 0 bits, but they're outside of the type range.

What are the bits outside of type size supposed to be?
I'm not sure who supervised #1623, @michael-schwarz, @jerhard? Does anyone know

Reply:
For now, I just wrapped things into a `Cil.fitsInInt`. Apparently, it wasn't just problematic for `_Bool`, but unsigned types in general, where the mask for definite leading zeroes (represented by some negative integer) doesn't really fit the type itself and truncates/wraps wrongly.
It's a bit similar to the issues around https://github.com/goblint/cil/pull/111.